app/models/user.rb

```diff
@@ -16,4 +17,23 @@ def self.digest string
             end
     BCrypt::Password.create string, cost: cost
   end
+
+  class << self
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute :remember_digest, User.digest(remember_token)
```

Comment:
Phân biệt `update_attribute` và `update_column` nhé 

```suggestion
    update_column :remember_digest, User.digest(remember_token)
```

Reply:
- update_attribute chỉ có thể update một attribute của object. Và nó có một số đặc điểm: Trường được update sẽ bị bỏ qua validate, Callback vẫn chạy, Trường updated_at vẫn được cập nhật
- update_column nó sẽ update trực tiếp vào DB và bỏ qua validate, bỏ qua callback, không cập nhật updated_at

Reply:
uk, nên chỗ này dùng update_column là hợp lý nhé